Release/include/cpprest/http_client.h

```diff
@@ -100,7 +100,6 @@ class http_client_config
 #endif
         , m_set_user_nativehandle_options([](native_handle)->void{})
 #if !defined(_WIN32) && !defined(__cplusplus_winrt)
-        , m_ssl_context_callback([](boost::asio::ssl::context&)->void{})
```

Comment:
If we initialize `m_ssl_context_callback` with an do-nothing lambda, then `http_client_config::get_ssl_context_callback()` always returns a `std::function` that has a valid target.  As the target of `std::function` is hard to compare, we are unable to reuse the connection pool.